packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ChangeFamily } from "../change-family";
+import { AnchorSet, Delta } from "../tree";
+
+export interface Commit<TChangeset> {
+    sessionId: SessionId;
+    seqNumber: SeqNumber;
+    refNumber: SeqNumber;
+    changeset: TChangeset;
+}
+
+type SessionId = number;
+type SeqNumber = number;
+
+/**
+ * Represents a local branch of a document and interprets the effect on the document of adding sequenced changes,
+ * which were based on a given session's branch, to the document history
+ */
+// TODO: Remove commits when they are no longer in the collab window
+// TODO: Try to reduce this to a single type parameter
+// TODO: Move logic into Rebaser if possible
+export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TChangeset>> {
+    private readonly trunk: Commit<TChangeset>[] = [];
+    private readonly branches: Map<SessionId, Branch<TChangeset>> = new Map();
+    private readonly localChanges: TChangeset[] = [];
+
+    public constructor(private readonly localSessionId: SessionId,
+        private readonly changeFamily: TChangeFamily,
+    ) { }
+
+    public addSequencedChange(newCommit: Commit<TChangeset>, anchors?: AnchorSet): Delta.Root {
```

Comment:
if anchors are provided, how are they updated? From old head of local to head of local? From the location on the local branch the commit was created at to after it in its new rebased position? From the old head of trunk to the new head of trunk?

Needs some docs, or a different API, to manage the anchorsSets.

I was thinking maybe we would just expose one public anchorSet (in a field, passed into the constructor) that is always kept at the head of the local branch, and update it as a side effect of these functions

Reply:
I made the AnchorSet public and added some logic to the testing code so that it checks the anchors are being rebased over the relevant changes.